### PR TITLE
A test of cascading nature of delete for all atlas entities

### PR DESCRIPTION
**Pull request by yazad3**
### Description:

Deletes in `Atlas` are cascading in nature, which means that when an entity is deleted other related entities are updated to maintain consistency. In some cases cascading is enforced manually by forcing the API user with errors to fix referential integrity issues and in other cases it is automatic.

Example, when am `entity` is deleted, it is also deleted as a `member` of any `relation` it may have belonged to.

~There are some potentially complex cases with with `node` and `relation` deletes which I am working on in a separate PR. This PR covers relatively simpler cases of cascade update to `member` list of any `relation` to which the deleted entity may have belonged to. Entities addressed in this PR include `point`, `line`, `edge`, and `area`.~

### Potential Impact:

None, most of the PR is just a test case.

### Unit Test Approach:

Unit Test to ensure expected behavior.

### Test Results:

Test successful.

------

In doubt: [Contributing Guidelines](CONTRIBUTING.md)

### CODE REVIEW

**Comment by yazad3** [comment]
@lucaspcram - I just merged my further changes for `Relation` and `Node` deletions in this PR.

(build seems to be stuck)

File: src/main/java/org/openstreetmap/atlas/geography/atlas/items/ConnectedEdgeType.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+/**
+ * {@link Node}s have in and out edges, this enum enables generalizing logic around this node and
+ * edge connection.
+ *
+ * @author Yazad Khambata
+ */
+@AllArgsConstructor
+@Getter
+public enum ConnectedEdgeType
```
Comment (lucaspcram) [inline src/main/java/org/openstreetmap/atlas/geography/atlas/items/ConnectedEdgeType.java]: This is a great idea. I may be able to update some code down the line.

File: src/main/java/org/openstreetmap/atlas/geography/atlas/items/Relation.java [hunk 2]
```diff
@@ -320,6 +322,30 @@ public boolean isMultiPolygon()
...
     public abstract RelationMemberList members();
 
+    /**
+     * Get a subset of {@link #members()} matching a certain {@link ItemType}.
+     *
+     * @param itemType
+     *            - the type of members to filter.
+     * @return - {@link #members()} of type itemType.
+     */
+    public RelationMemberList membersOfType(final ItemType itemType)
```
Comment (lucaspcram) [inline src/main/java/org/openstreetmap/atlas/geography/atlas/items/Relation.java]: One suggestion for this one: perhaps the parameter could be variadic for even more flexibility?
```
public RelationMemberList membersOfType(final ItemType... itemTypes)
```

File: src/test/java/org/openstreetmap/atlas/geography/atlas/change/CascadeDeleteTestHelper.java [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+import org.openstreetmap.atlas.geography.atlas.items.Node;
+import org.openstreetmap.atlas.geography.atlas.items.Relation;
+import org.openstreetmap.atlas.geography.atlas.items.RelationMember;
+
+/**
+ * @author Yazad Khambata
+ */
+class CascadeDeleteTestHelper
+{
+
```
Comment (lucaspcram) [inline src/test/java/org/openstreetmap/atlas/geography/atlas/change/CascadeDeleteTestHelper.java]: nit: extra space

File: src/main/java/org/openstreetmap/atlas/geography/atlas/items/ConnectedEdgeType.java (hunk 1 above)
Comment (yazad3) [reply src/main/java/org/openstreetmap/atlas/geography/atlas/items/ConnectedEdgeType.java]: Go for it :)

File: src/main/java/org/openstreetmap/atlas/geography/atlas/items/Relation.java (hunk 2 above)
Comment (yazad3) [reply src/main/java/org/openstreetmap/atlas/geography/atlas/items/Relation.java]: Done!

File: src/test/java/org/openstreetmap/atlas/geography/atlas/change/CascadeDeleteTestHelper.java (hunk 3 above)
Comment (yazad3) [reply src/test/java/org/openstreetmap/atlas/geography/atlas/change/CascadeDeleteTestHelper.java]: Done :)

**Pull request merged**